src/transformers/tokenization_utils_base.py

```diff
@@ -0,0 +1,1775 @@
+        padding: Union[bool, str] = False,
+        truncation: Union[bool, str] = False,
+        max_length: Optional[int] = None,
+        stride: int = 0,
+        return_tensors: Optional[Union[str, TensorType]] = None,
+        **kwargs
+    ):
+        """
+        Converts a string in a sequence of ids (integer), using the tokenizer and vocabulary.
+
+        Same as doing ``self.convert_tokens_to_ids(self.tokenize(text))``.
+
+        Args:
+            text (:obj:`str`, :obj:`List[str]` or :obj:`List[int]`):
+                The first sequence to be encoded. This can be a string, a list of strings (tokenized string using
+                the `tokenize` method) or a list of integers (tokenized string ids using the `convert_tokens_to_ids`
+                method)
+            text_pair (:obj:`str`, :obj:`List[str]` or :obj:`List[int]`, `optional`, defaults to :obj:`None`):
+                Optional second sequence to be encoded. This can be a string, a list of strings (tokenized
+                string using the `tokenize` method) or a list of integers (tokenized string ids using the
+                `convert_tokens_to_ids` method)
+        """
+        encoded_inputs = self.encode_plus(
+            text,
+            text_pair=text_pair,
+            add_special_tokens=add_special_tokens,
+            padding=padding,
+            truncation=truncation,
+            max_length=max_length,
+            stride=stride,
+            return_tensors=return_tensors,
+            **kwargs,
+        )
+
+        return encoded_inputs["input_ids"]
+
+    def num_special_tokens_to_add(self, pair: bool = False) -> int:
+        raise NotImplementedError
+
+    def _get_padding_truncation_strategies(
+        self, padding=False, truncation=False, max_length=None, verbose=True, **kwargs
+    ):
+        """ Find the correct padding/truncation strategy with backward compatibility
+            for old arguments (truncation_strategy and pad_to_max_length) and behaviors.
+        """
+        old_truncation_strategy = kwargs.pop("truncation_strategy", "do_not_truncate")
+        old_pad_to_max_length = kwargs.pop("pad_to_max_length", False)
+
+        # Backward compatibility for previous behavior, maybe we should deprecate it:
+        # If you only set max_length, it activates truncation for max_length
+        if max_length is not None and padding is False and truncation is False:
+            if verbose:
+                logger.warning(
+                    "Truncation was not explicitely activated but `max_length` is provided a specific value, "
+                    "please use `truncation=True` to explicitely truncate examples to max length. "
+                    "Defaulting to 'only_first' truncation strategy. "
+                    "If you encode pairs of sequences (GLUE-style) with the tokenizer you may want to check this is the right behavior."
+                )
+            truncation = "only_first"
```

Comment:
Just a pedantic comment, but this could be dangerous, because `truncation` is originally a boolean.

Falsy/Truthy comparison is done later with:

```python
elif truncation is not False:
            if truncation is True:
                truncation_strategy = (
                    TruncationStrategy.ONLY_FIRST
                )  # Default to truncate the first sequences in pairs of inputs
            else:
                truncation_strategy = TruncationStrategy(truncation)
```

and we will land in the ` truncation_strategy = TruncationStrategy(truncation)` branch.

But then `truncation_strategy` will be same as used in the ` if truncation is True` branch - so it is a kind of "hidden" duplicate code here 🤔